src/cohort/cohort_server.cc

```diff
@@ -1,56 +1,265 @@
 #include "src/cohort/cohort_server.h"
 
+#include <fstream>
+
 #include "absl/status/statusor.h"
+#include "absl/synchronization/mutex.h"
 #include "absl/time/time.h"
 #include "grpcpp/server_context.h"
+#include "src/blockchain/two_phase_commit.h"
 #include "src/proto/cohort.grpc.pb.h"
-#include "src/utils/grpc_utils.h"
+#include "src/utils/status_utils.h"
 
 namespace cohort {
 
 namespace {
 
+using Blockchain = ::blockchain::TwoPhaseCommit;
 using ::grpc::ServerContext;
 
-std::chrono::time_point<std::chrono::system_clock> FromProtoTimestamp(
-    const google::protobuf::Timestamp &timestamp) {
-  return absl::ToChronoTime(absl::FromUnixSeconds(timestamp.seconds()) +
-                            absl::Nanoseconds(timestamp.nanos()));
+Blockchain::VotingDecision WaitForBlockchainDecision(
+    const std::string& transaction_id, absl::Time presumed_abort_time) {
+  std::this_thread::sleep_until(absl::ToChronoTime(presumed_abort_time));
+  Blockchain::VotingDecision decision =
+      Blockchain::GetVotingDecision(transaction_id)
+          .value_or(Blockchain::VotingDecision::PENDING);
+  while (decision == Blockchain::VotingDecision::PENDING) {
```

Comment:
Do you want to print some logs (maybe next PR)? There are few while() { doX(); } logic which might be hard to debug without logs.

Reply:
+1 good idea. I'll do that in this PR.